plugins/plugin.go

```diff
@@ -151,39 +151,85 @@ func (p *Plugin) StartPlugin() error {
 	return p.App.Run(os.Args)
 }
 
-// Metrics polls the DC/OS components and returns a slice of
+// Metrics polls the local dcos-metrics API and returns a slice of
 // producers.MetricsMessage.
 func (p *Plugin) Metrics() ([]producers.MetricsMessage, error) {
+	var messages []producers.MetricsMessage
 
-	metricsMessages := []producers.MetricsMessage{}
+	// Fetch node metrics
+	nodeMetrics, err := p.getNodeMetrics()
+	if err != nil {
+		return messages, err
```

Comment:
I think it's more common to return `nil, err` in this situation unless there's a partial result.